Library/Homebrew/test/command_options_spec.rb

```diff
@@ -0,0 +1,53 @@
+require "command_options"
+
+describe Homebrew::CommandOptions do
+  describe "Tests for CommandOptions() class: " do
+    it "checks correctness of initialize() method" do
+      command_options = Homebrew::CommandOptions.new("test command")
+      expect(command_options.command_name).to eq("test command")
+      expect(command_options.valid_options).to eq({})
+    end
+
+    it "checks correct setting of valid_options" do
+      command_options = Homebrew::CommandOptions.new("test command")
+      command_options.option("--bar")
+      command_options.option("--foo", "do foo")
+      command_options.option("--quiet", "be quiet")
+      expect(command_options.valid_options).to eq("--foo"=>"do foo", "--quiet"=>"be quiet",
+        "--bar"=>"No description for this option is available")
+    end
+
+    it "checks error message if invalid option(s) are provided" do
```

Comment:
Can we split this into two tests? The shorter a test, the easier it is to understand, and the easier it is to pinpoint what's not working if one part of the test starts failing.